espresso/streamer.go

```diff
@@ -110,48 +129,72 @@ func (s *EspressoStreamer) Update(ctx context.Context) error {
 			return fmt.Errorf("failed to fetch transactions in block: %w", err)
 		}
 
+		s.Log.Info(fmt.Sprint("Fetched HS block %w", rune(s.hotShotPos)))
 		if len(txns.Transactions) == 0 {
-			s.Log.Debug("no transactions in hotshot block", "blockNr", s.hotShotPos)
+			s.Log.Info("no transactions in hotshot block", "blockNr", s.hotShotPos)
 			continue
+		} else {
+			s.Log.Warn("yes transactions in hotshot block", "blockNr", s.hotShotPos)
 		}
 
-		rawHeader, err := s.EspressoClient.FetchRawHeaderByHeight(ctx, s.hotShotPos)
-		if err != nil {
-			return fmt.Errorf("failed to fetch raw HotShot header: %w", err)
-		}
+		// rawHeader, err := s.EspressoClient.FetchRawHeaderByHeight(ctx, s.hotShotPos)
+		// if err != nil {
+		// 	return fmt.Errorf("failed to fetch raw HotShot header: %w", err)
+		// }
 
-		var header espressoTypes.HeaderImpl
-		err = json.Unmarshal(rawHeader, &header)
-		if err != nil {
-			return fmt.Errorf("could not unmarshal header from bytes")
-		}
+		// var header espressoTypes.HeaderImpl
+		// err = json.Unmarshal(rawHeader, &header)
+		// if err != nil {
+		// 	return fmt.Errorf("could not unmarshal header from bytes")
+		// }
 
-		snapshot, err := s.EspressoLightClient.FetchMerkleRoot(s.hotShotPos, nil)
-		if err != nil {
-			return fmt.Errorf("failed to fetch Merkle root: %w", err)
-		}
+		// snapshot, err := s.EspressoLightClient.FetchMerkleRoot(s.hotShotPos, nil)
+		// if err != nil {
+		// 	return fmt.Errorf("failed to fetch Merkle root: %w", err)
+		// }
 
-		if snapshot.Height <= s.hotShotPos {
-			return fmt.Errorf("snapshot height is less than or equal to the requested height")
-		}
+		// if snapshot.Height <= s.hotShotPos {
+		// 	return fmt.Errorf("snapshot height is less than or equal to the requested height")
+		// }
 
 		// TODO Philippe initialize when creating the streamer
-		s.BatchBuffer.SetBatcherAddress(s.BatcherAddress)
+		//s.BatchBuffer.SetBatcherAddress(s.BatcherAddress)
 		for _, transaction := range txns.Transactions {
 
-			s.BatchBuffer.SetBatchPos(s.BatchPos)
-			s.BatchBuffer.SetHeader(header)
-			s.BatchBuffer.ParseAndInsert(transaction)
+			batch, err := s.UnmarshalBatch(transaction)
+			if err != nil {
+				// Invalid Batch
+				s.Log.Warn("Invalid batch", "error", err)
+				continue
+			}
+
+			if (*batch).Number() < s.BatchPos {
+				continue
+			}
+
+			// origin := (*batch).L1Origin()
```

Comment:
Starts of an attempt to ensure sequencer proposed batch with finalized L1 state.
In general, this section of the code is unfinished, probably needs more checks for batch validity.

Reply:
Follow-up: https://app.asana.com/1/1208976916964769/project/1209392461754458/task/1209980878478927?focus=true